File: flink-pipeline-tasks/flink19-pipeline-task/src/main/java/com/turing/java/flink19/pipeline/demo2/FraudDetectionJob.java
package com.turing.java.flink19.pipeline.demo2;

import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.walkthrough.common.sink.AlertSink;
import org.apache.flink.walkthrough.common.entity.Alert;
import org.apache.flink.walkthrough.common.entity.Transaction;
import org.apache.flink.walkthrough.common.source.TransactionSource;

/**
 * @descr 欺诈检测
 *
 * @author Tony
 * @date 2024-07
 * */
public class FraudDetectionJob {

    public static void main(String[] args) throws Exception {

        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        DataStream<Transaction> transactions = env
                .addSource(new TransactionSource())
                .name("transactions");

        transactions.print();
        DataStream<Alert> alerts = transactions
                .keyBy(Transaction::getAccountId)
                .process(new FraudDetector())
                .name("fraud-detector");

        alerts.print();
        alerts.addSink(new AlertSink())
                .name("send-alerts");

        env.execute("Fraud Detection");
    }

}

File: flink-pipeline-tasks/flink19-pipeline-task/src/main/scala/com/turing/scala/flink19/pipeline/demo1/FraudDetectionJob.scala
package com.turing.scala.flink19.pipeline.demo1

import org.apache.flink.streaming.api.datastream.DataStream
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment
import org.apache.flink.walkthrough.common.entity
import org.apache.flink.walkthrough.common.entity.Alert
import org.apache.flink.walkthrough.common.source.TransactionSource

//程序的数据处理流
object FraudDetectionJob {

  def main(args: Array[String]): Unit = {

    val env=StreamExecutionEnvironment.createLocalEnvironment()

    val transactions:DataStream[entity.Transaction]=env
      .addSource(new TransactionSource)
      .name("transactions")

    transactions.print()
    // transactions.print()
//    val alerts: DataStream[Alert] = transactions
//      .keyBy(transaction => transaction.getAccountId)
//      .process(new FraudDetector)
//      .name("fraud-detector")

//    alerts.print()
    //      .addSink(new AlertSink)
    //      .name("send-alerts")

    env.execute("Fraud Detection")
  }

}
